packages/relay/src/lib/precheck.ts

```diff
@@ -169,10 +170,10 @@ export class Precheck {
     const passes = txGasPrice >= networkGasPrice || Precheck.isDeterministicDeploymentTransaction(tx);
```

Comment:
This was discussed offline with @Neurone to not force the Relay provider to set `GAS_PRICE_TINY_BAR_BUFFER` when `MAX_ALLOWANCE` is set.

```suggestion
    const passes = txGasPrice >= networkGasPrice || Precheck.isDeterministicDeploymentTransaction(tx) || (txGasPrice === 0n && ConfigService.get('MAX_ALLOWANCE') > 0;
```